bigquery_etl/glean_usage/baseline_clients_first_seen.py

```diff
@@ -0,0 +1,183 @@
+    "--date",
+    required=True,
+    type=lambda d: datetime.strptime(d, "%Y-%m-%d").date(),
+    help="Date partition to process, in format 2019-01-01",
+)
+parser.add_argument(
+    "--output_dir",
+    "--output-dir",
+    help="Also write the query text underneath the given sql dir",
+)
+parser.add_argument(
+    "--output_only",
+    "--output-only",
+    "--views_only",  # Deprecated name
+    "--views-only",  # Deprecated name
+    action="store_true",
+    help=(
+        "If set, we only write out sql to --output-dir and we skip"
+        " running the queries"
+    ),
+)
+standard_args.add_parallelism(parser)
+standard_args.add_dry_run(parser, debug_log_queries=False)
+standard_args.add_log_level(parser)
+standard_args.add_priority(parser)
+standard_args.add_billing_projects(parser)
+standard_args.add_table_filter(parser)
+
+
+TARGET_TABLE_ID = "baseline_clients_first_seen_v1"
+INIT_FILENAME = f"{TARGET_TABLE_ID}.init.sql"
+QUERY_FILENAME = f"{TARGET_TABLE_ID}.query.sql"
+VIEW_FILENAME = f"{TARGET_TABLE_ID[:-3]}.view.sql"
+VIEW_METADATA_FILENAME = f"{TARGET_TABLE_ID[:-3]}.metadata.yaml"
+
+
+def main():
+    """Generate and run queries based on CLI args."""
+    args = parser.parse_args()
+
+    try:
+        logging.basicConfig(level=args.log_level, format="%(levelname)s %(message)s")
+    except ValueError as e:
+        parser.error(f"argument --log-level: {e}")
+
+    baseline_tables = list_baseline_tables(
+        project_id=args.project_id,
+        only_tables=getattr(args, "only_tables", None),
+        table_filter=args.table_filter,
+    )
+
+    with ThreadPool(args.parallelism) as pool:
+        # Do a first pass with dry_run=True so we don't end up with a partial success;
+        # we also write out queries in this pass if so configured.
+        pool.map(
+            partial(
+                run_query,
+                args.project_id,
+                date=args.date,
+                dry_run=True,
+                output_dir=args.output_dir,
+                output_only=args.output_only,
+            ),
+            baseline_tables,
+        )
+        if args.output_only:
+            return
+        logging.info(
+            f"Dry runs successful for {len(baseline_tables)}"
+            " baseline_clients_first_seen table(s)"
+        )
+        # Now, actually run the queries.
+        if not args.dry_run:
+            pool.map(
+                partial(run_query, args.project_id, date=args.date, dry_run=False),
+                baseline_tables,
+            )
+
+
+def run_query(
+    project_id, baseline_table, date, dry_run, output_dir=None, output_only=False
+):
+    """Process a single table, potentially also writing out the generated queries."""
+    tables = table_names_from_baseline(baseline_table)
+
+    table_id = tables["first_seen_table"]
+    view_id = tables["first_seen_view"]
+    render_kwargs = dict(
+        header="-- Generated via bigquery_etl.glean_usage\n",
+        fennec_id=any(
```

Comment:
What's the status of Firefox for iOS in terms of having a ping that contains the legacy identifier? Do we know what that will look like and whether we'll be able to adapt this approach to handle iOS?

Reply:
I believe there hasn't been movement on this for quite some time (https://jira.mozilla.com/browse/FXIOS-1415), but this approach should be fine in the future with some minor modifications.